Add 'setuptools' as dependency for 'ferment'

antoineco opened the pull request:
Without it, the task "_Install ferment generator_" fails (tested on Ubuntu 16.04)

Before commit:

```
TASK [debops.docker : Install ferment generator] *******************************
fatal: [nelson.cluster]: FAILED! => {
  "changed": false,
  "cmd": "/usr/local/bin/pip install ferment",
  "failed": true, "msg":
  "stdout: Collecting ferment\n  Using cached Ferment-0.0.1-py2.py3-none-any.whl\nRequirement already satisfied (use --upgrade to upgrade): docker-py in /usr/lib/python2.7/dist-packages (from ferment)\nCollecting wheezy.template (from ferment)\n  Using cached wheezy.template-0.1.167.tar.gz\n\n:stderr: Could not import setuptools which is required to install from a source distribution.\nPlease install setuptools.\n"
}
```

After commit:

```
TASK [debops.docker : Install ferment generator] *******************************
changed: [nelson.cluster] => (item=setuptools)
changed: [nelson.cluster] => (item=ferment)
```

drybjed commented:
Good idea. Any chance to  switch from the PyPi package to the APT package (`python-setuptools`)? I'd like to stick to APT as much as possible. You can add it in `docker_base_packages` list.

antoineco commented:
Sure thing, thanks for the suggestion 👍 I just rebased my fork and fixed some deprecated stuff.

_off topic: do you accept contributions for new roles as well? The "Contributing guidelines" only mention existing roles. I love this project and would like to submit service roles for Hashicorp's Consul and Nomad_

antoineco committed:
Add python-setuptools apt package

antoineco committed:
Bare variables are deprecated

antoineco committed:
Fix error when 'ansible_ssh_user' is undefined

drybjed commented:
Excellent, thanks for fixting the other issues as well. I'll merge that in a bit.

Passing your own roles to DebOps is certaiinly possible and encouraged. :) I suppose that contributor documentation would need to be updated. In the meantime, what you should do is publish the roles on GitHub and point to them on `#debops` IRC channel or on the mailing list. After evaluation the repository can be moved to `debops` organization on GitHub and roles can be integrated with the rest.

Alternatively there's a https://github.com/debops-contrib/ organization where you could gain access if you plan to contribute more roles in the future. This organization is currently something like a stepping stone, I want to move the roles from there to the main DebOps organization when I get some time to integrate them properly. You will also get write access to it so publishing your roles is easier.

As for Hashicorp projects like Consul, i have mixed feelings. They look very interesting and could be great addition to the project, but from my point of view it's hard to install them without requiring some work on the user part. First, check out [my software policy for DebOps](https://wiki.debops.org/wiki:software-policy). In short - DebOps should use software from Debian Archive, or upstream, or local APT repositories set up by the user. At the moment Consul et al aren't available in Debian Archive and Hashicorp doesn't offer any kind of APT repositories - they just offer `.zip` packages (yeah, so at least `unzip` is a requirement). You could either download the packages directly to each host and handle them crudely without any automated authentication, or create your own `.deb` packages published via local APT repository.

So, I'm torn. Hashicorp doesn't seem to be interested in providing their own APT repository deeming it too much work, essentially burdening their users with it. Not sure when their products will be available in Debian Archive. What's your solution for this issue?

antoineco commented:
That's correct, I used the workaround you mentioned. The Consul/Nomad version is set in a variable. The archive is downloaded from Hashicorp's CDN and persisted somewhere on the local drive using `get_url` (with sha256 checksum validation), then extracted using `unarchive` to `/usr/local/bin` every time the previous condition changes (e.g. a new version has been downloaded).

It doesn't replace a repository of course, unfortunately. I'll see if I can push that to `debops-contrib`.

Anyway, thanks for reviewing my PR so quickly.

drybjed commented:
Sorry, I'm currently on vacation, usually I keep up to date quickier. :-) I suppose that `get_url` would be required. In that case package should be downloaded to `/usr/local/src/<subdir>/` for archival purposes. Using `unarchive` module seems reasonable as well, I guess `unzip` will need to be present on the host.

Publish your role in your own GitHub account first, then we can move it to DebOps organization after a review.
